src/services/deployment.ts

```diff
@@ -49,6 +49,18 @@ const isCompletedDeployment = (state: CompassDeploymentEventState) => {
   return state === CompassDeploymentEventState.Failed || state === CompassDeploymentEventState.Successful;
 };
 
+export const DESCRIPTION_TRUNCATION_LENGTH = 255;
+
+export const truncateProjectNameString = (beforeString: string, projectName: string, afterString: string) => {
+  // Spaces need to be included in `beforeString` and `afterString` so they can be considered in string length
+  let truncatedProjectName = projectName;
+  if (beforeString.length + projectName.length + afterString.length > DESCRIPTION_TRUNCATION_LENGTH) {
+    const projectNameLen = DESCRIPTION_TRUNCATION_LENGTH - beforeString.length - afterString.length;
+    truncatedProjectName = projectName.slice(0, projectNameLen);
+  }
+  return `${beforeString}${truncatedProjectName}${afterString}`;
```

Comment:
Could we move this function to `utils/event-mapping.ts` file since it is shared between builds/deployment? Then we can also create a `event-mapping.test.ts` file to add some unit tests for this function.

Reply:
sounds good! Here are the test cases I'm thinking of: 
- beforeString + projectName + afterString is short enough and doesn't need to be truncated
- ^ is too long and needs to be truncated to 255 characters 
- if it is possible, assert that in both scenarios an error is not thrown